### ISSUE TITLE
WH-B0: tilize_init/uninit skips packer reconfiguration for OCB tile geometry, breaking tiny-tile retile

### ISSUE THREAD

**Issue by mouliraj-mcw** [tenstorrent/tt-metal#52175]
## Summary

On Wormhole B0, `ckernel::tilize_init(icb, block, ocb)` / `ckernel::tilize_uninit(icb, ocb)` (in `tt_metal/hw/inc/api/compute/tilize.h`) do not reconfigure the packer for the OCB's tile geometry. As a result, retile / tilize kernels that pack into an output CB whose `face_r_dim` differs from the pack's initial configuration silently emit wrong bytes for the affected tile-rows.

This is the reason the `test_tilize_retile` / `test_tilize_row_major_to_tiny_tile` families in `tests/ttnn/unit_tests/operations/data_movement/test_tilize.py` are `@skip_for_wormhole_b0` — the WH LLK primitive can support tiny tiles, but the `tilize_init/uninit` API doesn't wire it in.

Discovered while adding a regression test for #51213 (item 5): the shrink-retile OOB-writer fix (PR-2 in that thread) is correct, but the WH retile path itself is functionally wrong for output tile heights ≠ the input tile's `face_r_dim`.

## Root cause

`tt_metal/hw/inc/api/compute/tilize.h`:

```cpp
ALWI void tilize_init(uint32_t icb, uint32_t block, uint32_t ocb, ...) {
#ifndef ARCH_QUASAR
    state_configure<Operand::SRCA, Operand::PACK>(icb, ocb, call_line);
    UNPACK((llk_unpack_tilize_init(icb, block)));
    MATH((llk_math_eltwise_unary_datacopy_init<..., PackMode::Tilize>(icb)));
#ifdef ARCH_BLACKHOLE
    PACK((llk_pack_init<PackMode::Tilize, false>(ocb, 1, icb)));
#endif                                     // ← nothing for ARCH_WORMHOLE
#else
    ...
#endif
}

ALWI void tilize_uninit(uint32_t icb, uint32_t ocb) {
    UNPACK((llk_unpack_tilize_uninit(icb)));
#ifdef ARCH_BLACKHOLE
    PACK((llk_pack_init<PackMode::Default>(ocb)));
#endif                                     // ← nothing for ARCH_WORMHOLE
}
```

The BH branch's `llk_pack_init(ocb, ...)` reads `face_r_dim`, `num_faces`, `partial_face`, `narrow_tile` from the OCB's CB metadata (via `get_output_face_r_dim(output_id)` etc. in `tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_pack_tile_api.h`) and reprograms the packer's MOP + ADCXX + addrmod. On WH-B0 this reconfiguration is simply absent — the packer stays in whatever geometry was programmed by `compute_kernel_hw_startup(src_cb, mid_cb)` (namely `face_r_dim = FACE_R_DIM = 16`, taken from `mid_cb`).

## Reproducer

`ttnn.tilize` retile path (`TilizeMultiCoreRetileProgramFactory`) using compute kernel `ttnn/cpp/ttnn/operations/data_movement/tilize/device/kernels/compute/retile.cpp`. Standalone Python (WH-B0):

```python
import torch, ttnn
device = ttnn.open_device(device_id=0)
cases = [
    ([1, 1, 16, 128], (32, 32), (16, 32)),  # baseline
    ([1, 1, 16, 128], (32, 32), (8,  32)),  # bug
    ([1, 1, 8,  128], (32, 32), (4,  32)),  # bug
]
for shape, in_tile, out_tile in cases:
    torch.manual_seed(42)
    x = torch.rand(shape, dtype=torch.bfloat16)
    tt_in  = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                             device=device, tile=ttnn.Tile(list(in_tile)))
    tt_out = ttnn.tilize(tt_in, tile=ttnn.Tile(list(out_tile)))
    got = ttnn.to_torch(tt_out)
    diffs = int((x != got).sum().item())
    print(f"in={in_tile}  out={out_tile}: diffs = {diffs}/{x.numel()}")
```

Result on WH-B0 (`origin/main`):

```
in=(32, 32)  out=(16, 32): diffs = 0/2048        ← passes (matches default face_r_dim=16)
in=(32, 32)  out=(8, 32) : diffs = 1023/2048     ← 2nd output tile-row entirely wrong
in=(32, 32)  out=(4, 32) : diffs = 1024/2048     ← wrong
```

For the `out(8,32)` case specifically, the corruption pattern is clean:

```
row  0..7 : 0 diffs
row  8..15: 128 diffs each (128 = full row width in bfloat16)
```

That is, the first output tile-row (`r=0` in the retile compute-kernel inner loop) is correct because the untilized rows 0..7 of `mid_view_cb` happen to align with what a `face_r_dim=16` packer emits; the second output tile-row (`r=1`) reads at `block_rd_ptr + words_per_out_tile_row` in `mid_view_cb` and the packer's row-stride vs the unpacker's X-dim counter no longer line up.

## Minimal patch and why it isn't enough

Attempted locally in `tt_metal/hw/inc/api/compute/tilize.h`:

```diff
 #ifdef ARCH_BLACKHOLE
     PACK((llk_pack_init<PackMode::Tilize, false>(ocb, 1, icb)));
+#elif defined(ARCH_WORMHOLE)
+    PACK((llk_pack_hw_configure<DST_ACCUM_MODE>(ocb)));
+    PACK((llk_pack_init<PackMode::Default>(ocb)));
 #endif
```

(and the symmetric `PACK((llk_pack_init<PackMode::Default>(ocb)))` restore in `tilize_uninit`). Re-run:

```
in=(32, 32)  out=(16, 32): PASS  (unchanged)
in=(32, 32)  out=(4, 32) : PASS  (fixed)
in=(32, 32)  out=(8, 32) : FAIL  (rows 8..15 still wrong)
```

The remaining `out(8,32)` failure is not on the pack side — the corruption is that the *unpacker's* X-dim counter (`TT_SETADCXX(p_setadc::PAC, pack_x_dim - 1, ...)` at `tt_llk_wormhole_b0/llk_lib/llk_pack.h:424`) is derived from `FACE_R_DIM` when `pack_mode != Untilize`, and its row-stride no longer matches the packer's per-tile row-stride once the packer is reprogrammed for a tiny OCB.

## Suggested full fix scope

1. In `tilize.h` (WH branch of `tilize_init`): plumb the OCB's `face_r_dim` and `num_faces` into a matching `_llk_unpack_tilize_init_` variant so the unpacker's ADCXX counter and the packer's per-tile stride agree.
2. Symmetric restore in `tilize_uninit` — `_llk_pack_init_<PackMode::Default>(icb)` + unpack ADCXX reset.
3. `_llk_pack_init_` on WH gates on `PackMode::Default | PackMode::Untilize` (see `static_assert` in `tt_llk_wormhole_b0/llk_lib/llk_pack.h:402`). No new PackMode variant is required — `PackMode::Default` with the OCB-derived `face_r_dim/num_faces` is sufficient.
4. Blast radius: every op that pairs `tilize_init` with an OCB whose tile shape differs from the primary OCB set at `compute_kernel_hw_startup` — `tilize_block`, `unpack_tilizeA_B_block`, retile, and any user-authored kernel following the same pattern. Wider regression sweep across `tilize`, `pack_untilize`, `transpose_wh`, matmul-fused tilize needs to run on WH after the change.

## Env

- Silicon: Wormhole B0 (`Arch.WORMHOLE_B0`)
- Base: `origin/main` at time of report

## Related

- #51213 (item 5) — the DRAM OOB writer bug PR-2 in that thread targets is orthogonal to this LLK gap. PR-2's regression test is `@skip_for_wormhole_b0` for the reason above.
